File: Ruby/lib/flip_and_invert_image.rb
class FlipAndInvertImage
    def self.flip_and_invert_image(a)
        result = []
        i = 0
        while a.length() > i do
            reversedArray = a[i].reverse()
            j = 0
            while reversedArray.length() > j do
                reversedArray[j] = reversedArray[j] === 1 ? 0 : 1
                j += 1
            end
            result.append(reversedArray)
            i += 1
        end
        result
    end
end

File: Javascript/flipAndInvertImage.js
const flipAndInvertImage = (arr) => {

    let result = [];
    for(let i = 0; i < arr.length; i++) {
        let reversedArray = arr[i].reverse();
        for(let j = 0; j < reversedArray.length; j++) {
            reversedArray[j] = reversedArray[j] === 1 ? 0 : 1;
        }
        result.push(reversedArray);
    }
    return result;
}

module.exports = flipAndInvertImage